src/opensearch_sdk_py/actions/internal/environment_settings_response_handler.py

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+
+import logging
+
+from opensearch_sdk_py.actions.request_response_handler import RequestResponseHandler
+from opensearch_sdk_py.actions.response_handler import ResponseHandler
+from opensearch_sdk_py.transport.environment_settings_response import EnvironmentSettingsResponse
+from opensearch_sdk_py.transport.outbound_message_request import OutboundMessageRequest
+from opensearch_sdk_py.transport.stream_input import StreamInput
+from opensearch_sdk_py.transport.stream_output import StreamOutput
+
+
+class EnvironmentSettingsResponseHandler(ResponseHandler):
+    def __init__(self, next_handler: RequestResponseHandler) -> None:
```

Comment:
You should probably sink `next_handler` into either the response handler (or a new class), or manage the sequence of handlers outside of the handler itself.

Reply:
> You should probably sink `next_handler` into either the response handler (or a new class), or manage the sequence of handlers outside of the handler itself.

100% agree, primarily with the "outside of the handler itself" part. I'm convinced there's got to be a better, more general way to "wait for this set of responses to complete".

Related, `RequestHandler` and `ResponseHandler` have differing arguments for `send()` but they ought to be the same.